flang/docs/DoConcurrentConversionToOpenMP.md

```diff
@@ -0,0 +1,240 @@
+
+The above also has the consequence that the `j` variable will **not** privatized in the OpenMP parallel/target region. In other words, it will be treated as if it was a `shared` variable. For more details about privatization, see the data environment section below.
+
+### Data environment
+
+By default, variables that are used inside a `do concurernt` loop nest are either treated as `shared` in case of mapping to `host`, or mapped into the `target` region using a `map` clause in case of mapping to `device`. The only exception to this is the loop's iteration variable(s) of **perfect** loop nest. In that case, for each IV, we allocate a local copy as shown the by the mapping examples above.
+
+#### Non-perfectly-nested loops' IVs
+
+For non-perfectly-nested loops, the IV(s) are still treated as `shared` or `map` entries as pointed out above. This **might not** be consistent with what the Fortran specficiation tells us. In particular, taking the following snippets from the spec (version 2023) into account:
+
+> § 3.35
+> ------
+> construct entity
+> entity whose identifier has the scope of a construct
+
+> § 19.4
+> ------
+>  A variable that appears as an index-name in a FORALL or DO CONCURRENT construct, or ... is a construct entity. A variable that has LOCAL or LOCAL_INIT locality in a DO CONCURRENT construct is a construct entity.
+> ...
+> The name of a variable that appears as an index-name in a DO CONCURRENT construct, FORALL statement, or FORALL construct has a scope of the statement or construct. A variable that has LOCAL or LOCAL_INIT locality in a DO CONCURRENT construct has the scope of that construct.
+
+From the above quotes, it seems there is an equivalence between the IV of a `do concurrent` loop and a variable with a `LOCAL` locality specifier (equivalent to OpenMP's `private` clause). Which means that we should probably localize/privatize a `do concurernt` loop's IV even if it is not perfectly nested in the nest we are parallelizing. For now, however, we **do not** that as pointed out previously. In the near future, we propose a middle-ground solution (see the Next steps section for more details).
+
+## Next steps
+
+### Delayed privatization
+
+So far, we emit the privatization logic for IVs inline in the paralle/target region. This is enough for out purposes right now since we don't localize/privatize any sophisticated types of variables yet. Once we have need for more advanced localization thought `do concurrent`'s locality specified (see below), delayed privatization will enable us to have a much cleaner IR. Once delayed privatization's implementation upstream is supported for the required constructs by the pass, we will move to it rather than inlined/early privatization.
```

Comment:
```suggestion
So far, we emit the privatization logic for IVs inline in the parallel/target region. This is enough for our purposes right now since we don't localize/privatize any sophisticated types of variables yet. Once we have need for more advanced localization through `do concurrent`'s locality specifiers (see below), delayed privatization will enable us to have a much cleaner IR. Once delayed privatization's implementation upstream is supported for the required constructs by the pass, we will move to it rather than inlined/early privatization.
```